src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs

```diff
@@ -594,34 +587,28 @@ protected virtual void OnCustomCommand(int command)
         ///       contains multiple associated services. Loads the specified services into memory so they can be
         ///       started.</para>
         /// </devdoc>
-        public static void Run(ServiceBase[] services)
+        public static unsafe void Run(ServiceBase[] services)
         {
             if (services == null || services.Length == 0)
                 throw new ArgumentException(SR.NoServices);
 
-            int sizeOfSERVICE_TABLE_ENTRY = Marshal.SizeOf<SERVICE_TABLE_ENTRY>();
-
-            IntPtr entriesPointer = Marshal.AllocHGlobal(checked((services.Length + 1) * sizeOfSERVICE_TABLE_ENTRY));
+            IntPtr entriesPointer = Marshal.AllocHGlobal(checked((services.Length + 1) * Marshal.SizeOf<SERVICE_TABLE_ENTRY>()));
```

Comment:
```suggestion
            IntPtr entriesPointer = Marshal.AllocHGlobal(checked((services.Length + 1) * sizeof(SERVICE_TABLE_ENTRY)));
```

It is not right to be combining `Marshal.SizeOf` with the manual marshaling. Also, `sizeof` is much faster.